src/js/tabs/ports.js

```diff
@@ -399,10 +402,68 @@ TABS.ports.initialize = function (callback, scrollPosition) {
                 blackbox_baudrate: blackboxBaudrate,
                 identifier: serialPort.identifier,
             };
+
             FC.SERIAL_CONFIG.ports.push(serialPortConfig);
+
+            // enable / disable features based on port configuration
+            serialPortConfig.functions.forEach(e => {
+                if (e === 'RX_SERIAL') {
+                    enableRxSerial = true;
+                }
+
+                if (e.startsWith("TELEMETRY")) {
+                    enableTelemetry = true;
+                }
+            });
+
+            if (serialPortConfig.functions.indexOf('BLACKBOX') >= 0) {
+                enableBlackbox = true;
+            }
+
+            if (serialPortConfig.functions.indexOf('ESC_SENSOR') >= 0) {
+                enableEsc = true;
+            }
+
+            if (serialPortConfig.functions.indexOf('GPS') >= 0) {
+                enableGps = true;
+            }
```

Comment:
Something like that woudl work as well
```suggestion
            let functions = serialPortConfig.functions
            enableRxSerial = functions.includes('RX_SERIAL');
            enableTelemetry = functions.some(e => e.startsWith("TELEMETRY"));
            enableBlackbox = functions.includes('BLACKBOX');
            enableEsc = functions.includes('ESC_SENSOR');
            enableGps = functions.includes('GPS');
```

Reply:
Thanks. Much easier to read.
Also @McGiverGim has a good point here.

Reply:
@McGiverGim yes would make sense. Have to look at it

Reply:
Also removed the promises.